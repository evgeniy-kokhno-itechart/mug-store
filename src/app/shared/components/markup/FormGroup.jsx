import React from 'react';
import { PropTypes } from 'prop-types';

const FormGroup = ({
  name, label, error, groupClasses, labelClasses, children,
}) => (
  <div className={`form-group ${groupClasses}`}>
    <label htmlFor={name} className={labelClasses}>{label}</label>
    {children}
    {error && <div className="alert alert-danger">{error}</div>}
  </div>
);

FormGroup.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  groupClasses: PropTypes.string,
  labelClasses: PropTypes.string,
  children: PropTypes.shape({
    type: PropTypes.string,
  }).isRequired,
};

FormGroup.defaultProps = {
  error: '',
  groupClasses: '',
  labelClasses: '',
};

export default FormGroup;
