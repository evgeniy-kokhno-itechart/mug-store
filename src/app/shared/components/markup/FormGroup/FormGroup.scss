@import 'Variables';

.formgroup {
    margin: 0.5rem 0;

    &__label {
        display: block;
        font-weight: 500;
        margin: 0.5rem 0.3rem;
    }

    &__error {
        background-color: rgba($color-danger-light, 0.5);
        border: 2px solid $color-danger-light;
        border-radius: 0 0 0.3rem 0.3rem;
        padding: 0.5rem;
    }
}