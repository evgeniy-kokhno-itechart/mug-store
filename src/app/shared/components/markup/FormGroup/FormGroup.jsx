import React from 'react';
import { PropTypes } from 'prop-types';
import './FormGroup.scss';

const FormGroup = ({
  name, label, error, groupClasses, labelClasses, children,
}) => (
  <div className={`formgroup ${groupClasses}`}>
    <label htmlFor={name} className={`formgroup__label ${labelClasses}`}>{label}</label>
    {children}
    {error && <div className="formgroup__error">{error}</div>}
  </div>
);

FormGroup.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  groupClasses: PropTypes.string,
  labelClasses: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

FormGroup.defaultProps = {
  error: '',
  groupClasses: '',
  labelClasses: '',
};

export default FormGroup;
