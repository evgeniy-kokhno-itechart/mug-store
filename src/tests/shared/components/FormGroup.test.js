/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormGroup from '../../../app/shared/components/markup/FormGroup/FormGroup';

configure({ adapter: new Adapter() });

describe('<FormGroup />', () => {
  let FormGroupWrapper;

  beforeEach(() => {
    FormGroupWrapper = shallow(
      <FormGroup
        name='testname'
        label='Test label'
      >
        <span>test child</span>
      </FormGroup>,
    );
  });

  it('renders properly with default props', () => {
    expect(FormGroupWrapper).toMatchSnapshot();
  });

  it('renders properly with error', () => {
    FormGroupWrapper.setProps({ error: 'test error' });
    expect(FormGroupWrapper).toMatchSnapshot();
  });

  it('renders properly with labelClasses', () => {
    FormGroupWrapper.setProps({ labelClasses: 'testlabelclass' });
    expect(FormGroupWrapper).toMatchSnapshot();
  });

  it('renders properly with groupClasses', () => {
    FormGroupWrapper.setProps({ groupClasses: 'testgroupclass' });
    expect(FormGroupWrapper).toMatchSnapshot();
  });
});
