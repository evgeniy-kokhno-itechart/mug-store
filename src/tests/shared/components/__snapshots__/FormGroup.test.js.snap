// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> renders properly with default props 1`] = `
<div
  className="form-group "
>
  <label
    className=""
    htmlFor="testname"
  >
    Test label
  </label>
  <span>
    test child
  </span>
</div>
`;

exports[`<FormGroup /> renders properly with error 1`] = `
<div
  className="form-group "
>
  <label
    className=""
    htmlFor="testname"
  >
    Test label
  </label>
  <span>
    test child
  </span>
  <div
    className="alert alert-danger"
  >
    test error
  </div>
</div>
`;

exports[`<FormGroup /> renders properly with groupClasses 1`] = `
<div
  className="form-group testgroupclass"
>
  <label
    className=""
    htmlFor="testname"
  >
    Test label
  </label>
  <span>
    test child
  </span>
</div>
`;

exports[`<FormGroup /> renders properly with labelClasses 1`] = `
<div
  className="form-group "
>
  <label
    className="testlabelclass"
    htmlFor="testname"
  >
    Test label
  </label>
  <span>
    test child
  </span>
</div>
`;
